pandas/tseries/offsets.py

```diff
@@ -572,6 +575,11 @@ def __getstate__(self):
 
     def __setstate__(self, state):
         """Reconstruct an instance from a pickled state"""
+        if 'offset' in state:
+            # Older versions have offset attribute instead of _offset
+            assert '_offset' not in state, list(state.keys())
+            state['_offset'] = state['offset']
```

Comment:
``state['_offset'] = state.pop('offset')``

can you remove the assert?

Reply:
Good call on the `pop`.  Change assert to ValueError if both keys are (somehow) there?

Reply:
you may want to add a 0.20.3 pickle that adds things for every frequency.

in separate PR.
you can use ``pandas/tests/io/generate_legacy_storage_files.py``, update to add all of the offsets. Then generate and add to the repo. All tests should pass.

Reply:
I'll make a note to do this once other fixes to offsets are done.